packages/babel-types/src/definitions/flow.js

```diff
@@ -1,136 +1,156 @@
 // @flow
 import defineType, {
-  assertEach,
-  assertNodeType,
+  arrayOfType,
+  assertOneOf,
   assertValueType,
-  chain,
+  validate,
+  validateOptional,
+  validateOptionalType,
+  validateType,
 } from "./utils";
 
 defineType("AnyTypeAnnotation", {
   aliases: ["Flow", "FlowBaseAnnotation"],
-  fields: {
-    // todo
-  },
 });
 
 defineType("ArrayTypeAnnotation", {
   visitor: ["elementType"],
   aliases: ["Flow"],
   fields: {
-    // todo
+    elementType: validateType("Flow"),
   },
 });
 
 defineType("BooleanTypeAnnotation", {
   aliases: ["Flow", "FlowBaseAnnotation"],
-  fields: {
-    // todo
-  },
 });
 
 defineType("BooleanLiteralTypeAnnotation", {
   aliases: ["Flow"],
-  fields: {},
+  fields: {
+    value: validate(assertValueType("boolean")),
+  },
 });
 
 defineType("NullLiteralTypeAnnotation", {
   aliases: ["Flow", "FlowBaseAnnotation"],
-  fields: {},
 });
 
 defineType("ClassImplements", {
   visitor: ["id", "typeParameters"],
   aliases: ["Flow"],
   fields: {
-    // todo
+    id: validateType("Identifier"),
+    typeParameters: validateOptionalType("TypeParameterInstantiation"),
   },
 });
 
 defineType("DeclareClass", {
   visitor: ["id", "typeParameters", "extends", "body"],
   aliases: ["Flow", "FlowDeclaration", "Statement", "Declaration"],
   fields: {
-    // todo
+    id: validateType("Identifier"),
+    typeParameters: validateOptionalType("TypeParameterInstantiation"),
+    extends: validateOptional(arrayOfType("InterfaceExtends")),
+    mixins: validateOptional(arrayOfType("InterfaceExtends")),
+    body: validateType("ObjectTypeAnnotation"),
   },
 });
 
 defineType("DeclareFunction", {
   visitor: ["id"],
   aliases: ["Flow", "FlowDeclaration", "Statement", "Declaration"],
   fields: {
-    // todo
+    id: validateType("Identifier"),
+    predicate: validateOptionalType("DeclaredPredicate"),
   },
 });
 
 defineType("DeclareInterface", {
   visitor: ["id", "typeParameters", "extends", "body"],
   aliases: ["Flow", "FlowDeclaration", "Statement", "Declaration"],
   fields: {
-    // todo
+    id: validateType("Identifier"),
+    typeParameters: validateOptionalType("TypeParameterDeclaration"),
+    extends: validateOptionalType("InterfaceExtends"),
+    mixins: validateOptional(arrayOfType("Flow")),
+    body: validateType("ObjectTypeAnnotation"),
   },
 });
 
 defineType("DeclareModule", {
-  visitor: ["id", "body"],
+  visitor: ["id", "body", "kind"],
```

Comment:
> Let's add `kind` to visitor list so we can do something like:
>
> `t.declareModule(t.identifier('d'), t.blockStatement([]), "ES")`

IIRC, the parameters are defined in the `builder` array. `visitor` should only contain the nodes which should be traversed.